documentation/upgrade_with_legacy_compilers.md

```diff
@@ -0,0 +1,47 @@
+# Upgrade Puppet Enterprise with legacy compilers
```

Comment:
Wondering if it makes more sense to make this a general legacy compilers doc instead of just upgrade with legacy compilers